File: services/finance-service/Messaging/Events/OrderCreatedEvent.cs
using System.Text.Json.Serialization;

namespace FinanceService.Messaging.Events;

public sealed class OrderCreatedEvent
{
    [JsonPropertyName("EventId")]
    public Guid EventId { get; set; }

    [JsonPropertyName("EventType")]
    public string EventType { get; set; } = "order.created";

    [JsonPropertyName("OrderId")]
    public string OrderId { get; set; } = string.Empty;

    [JsonPropertyName("CustomerAccountId")]
    public string CustomerAccountId { get; set; } = string.Empty;

    [JsonPropertyName("RestaurantId")]
    public string RestaurantId { get; set; } = string.Empty;

    [JsonPropertyName("BranchId")]
    public string BranchId { get; set; } = string.Empty;

    [JsonPropertyName("Subtotal")]
    public decimal Subtotal { get; set; }

    [JsonPropertyName("Total")]
    public decimal Total { get; set; }

    [JsonPropertyName("Status")]
    public string Status { get; set; } = string.Empty;

    [JsonPropertyName("CreatedAt")]
    public DateTime CreatedAt { get; set; }

    [JsonPropertyName("OccurredAt")]
    public DateTime OccurredAt { get; set; }
}

File: services/notification-service/src/main/java/com/tap2eat/notification/messaging/events/OrderCreatedEvent.java
package com.tap2eat.notification.messaging.events;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;

public record OrderCreatedEvent(
        @JsonProperty("EventId") String eventId,
        @JsonProperty("EventType") String eventType,
        @JsonProperty("OrderId") String orderId,
        @JsonProperty("CustomerAccountId") String customerAccountId,
        @JsonProperty("RestaurantId") String restaurantId,
        @JsonProperty("BranchId") String branchId,
        @JsonProperty("Subtotal") BigDecimal subtotal,
        @JsonProperty("Total") BigDecimal total,
        @JsonProperty("Status") String status,
        @JsonProperty("CreatedAt") String createdAt,
        @JsonProperty("OccurredAt") String occurredAt
) {
}
